templates/apigee-entity-default.html.twig

```diff
@@ -0,0 +1,80 @@
+ *
+ * @see \Drupal\apigee_edge\Entity\EdgeEntityViewBuilder::getBuildDefaults()
+ * @see \Drupal\Core\Entity\EntityViewBuilder::getBuildDefaults()
+ *
+ * If you need an entity type specific template, do not override this.
+ * Register an entity type specific theme in hook_theme() and implement your
+ * own template_preprocess_ENTITY_TYPE() template preprocess, template file and
+ * hook_theme_suggestions_ENTITY_TYPE() if needed.
+ *
+ * @code
+ *   function MY_MODULE_theme() {
+ *     return [
+ *       'developer_app' => [
+ *          'render element' => 'elements',
+ *       ],
+ *     ];
+ *   }
+ *
+ *   function template_preprocess_developer_app(&$variables) {};
+ *
+ *   function MY_MODULE_theme_suggestions_developer_app(array $variables) {}
+ * @endcode
+ *
+ * If you would like to re-use the implementation from
+ * template_preprocess_apigee_entity_default() you can call this function in
+ * your own entity type specific template preprocess function or add it to the
+ * entity type specific preprocess functions in hook_theme_registry_alter().
+ *
+ * Available variables:
+ * - entity: The entity with limited access to object properties and methods.
+ *   Only method names starting with "get", "has", or "is" and a few common
+ *   methods such as "id", "label", and "bundle" are available. For example:
+ *   - entity.getEntityTypeId() will return the entity type ID.
+ *   - entity.hasField('field_example') returns TRUE if the entity includes
+ *     field_example. (This does not indicate the presence of a value in this
+ *     field.)
+ *   Calling other methods, such as entity.delete(), will result in an exception.
+ *   See \Drupal\apigee_edge\Entity\EdgeEntityInterface for a full list of
+ *   methods.
+ * - label: The title of the entity.
+ * - content: All rendered field items. Use {{ content }} to print them all,
+ *   or print a subset such as {{ content.field_example }}. Use
+ *   {{ content|without('field_example') }} to temporarily suppress the printing
+ *   of a given child element.
+ * - url: Direct URL of the current entity.
+ * - attributes: HTML attributes for the containing element.
+ * - title_attributes: Same as attributes, except applied to the main title
+ *   tag that appears in the template.
+ * - content_attributes: Same as attributes, except applied to the main
+ *   content tag that appears in the template.
+ * - title_prefix: Additional output populated by modules, intended to be
+ *   displayed in front of the main title tag that appears in the template.
+ * - title_suffix: Additional output populated by modules, intended to be
+ *   displayed after the main title tag that appears in the template.
+ * - view_mode: View mode; for example, "teaser" or "full".
+ *
+ * @see template_preprocess_apigee_entity_default()
+ *
+ * @ingroup themeable
+ */
+#}
+<article{{ attributes }}>
+
+  {{ title_prefix }}
+  {% if label and view_mode != 'full' %}
+    <h2{{ title_attributes }}>
```

Comment:
Can we make the element configurable from preprocess instead of hardcoding `<h2>`?

Reply:
This is a bare minimum, default template based on the node.html.twig. If you would like to make it customizable, create a new entity type specific theme and ENTITY_TYPE.htm.yml in your theme/module. :)

Reply:
We constantly modify the node template for teaser mode because `<h2>` is almost never good. That's why I suggested to make it configurable from preprocess, which is less code than adding a new template. But I'm also ok with the original solution if you want something similar to `node.html.twig`.